packages/client/hmi-client/src/components/widgets/tera-chart-settings.vue

```diff
@@ -118,7 +162,23 @@ const emits = defineEmits([
 ]);
 
 // Settings of the same type that we want to interact with.
-const targetSettings = computed(() => props.settings.filter((s) => s.type === props.type));
+const targetSettings = computed(() => {
+	if (props.type === ChartSettingType.VARIABLE_COMPARISON) {
+		// For comparison charts, we only want to show the settings items for the current setting
+		return props.settings.filter((s) => s.id === props.selectedOptions[0]);
+	}
+	// For other chart types, keep existing behavior
+	return props.settings.filter((s) => s.type === props.type);
+});
```

Comment:
It looks like `targetSettings` isn't used anywhere in the comprison chart related template above. 
Do we need these changes? 

```suggestion
const targetSettings = computed(() => props.settings.filter((s) => s.type === props.type));
```